openstreetmap-carto.lua

```diff
@@ -0,0 +1,518 @@
+        tertiary        = {z = 340, roads = false},
+        residential     = {z = 330, roads = false},
+        unclassified    = {z = 330, roads = false},
+        road            = {z = 330, roads = false},
+        living_street   = {z = 320, roads = false},
+        pedestrian      = {z = 310, roads = false},
+        raceway         = {z = 300, roads = false},
+        motorway_link   = {z = 240, roads = true},
+        trunk_link      = {z = 230, roads = true},
+        primary_link    = {z = 220, roads = true},
+        secondary_link  = {z = 210, roads = true},
+        tertiary_link   = {z = 200, roads = false},
+        service         = {z = 150, roads = false},
+        track           = {z = 110, roads = false},
+        path            = {z = 100, roads = false},
+        footway         = {z = 100, roads = false},
+        bridleway       = {z = 100, roads = false},
+        cycleway        = {z = 100, roads = false},
+        steps           = {z = 90,  roads = false},
+        platform        = {z = 90,  roads = false},
+        construction    = {z = 10,  roads = false}
+    },
+    railway = {
+        rail            = {z = 440, roads = true},
+        subway          = {z = 420, roads = true},
+        narrow_gauge    = {z = 420, roads = true},
+        light_rail      = {z = 420, roads = true},
+        funicular       = {z = 420, roads = true},
+        preserved       = {z = 420, roads = false},
+        monorail        = {z = 420, roads = false},
+        miniature       = {z = 420, roads = false},
+        turntable       = {z = 420, roads = false},
+        tram            = {z = 410, roads = false},
+        disused         = {z = 400, roads = false},
+        construction    = {z = 400, roads = false},
+        platform        = {z = 90,  roads = false},
+    },
+    aeroway = {
+        runway          = {z = 60,  roads = false},
+        taxiway         = {z = 50,  roads = false},
+    },
+    boundary = {
+        administrative  = {z = 0,  roads = true}
+    },
+}
+
+local excluded_railway_service = {
+    spur = true,
+    siding = true,
+    yard = true
+}
+--- Gets the z_order for a set of tags
+-- @param tags OSM tags
+-- @return z_order if an object with z_order, otherwise nil
+function z_order(tags)
+    local z = 0
+    for k, v in pairs(tags) do
+        if roads_info[k] and roads_info[k][v] then
+            z = math.max(z, roads_info[k][v].z)
+        end
+    end
+    return z ~= 0 and z or nil
+end
+
+--- Gets the roads table status for a set of tags
+-- @param tags OSM tags
+-- @return 1 if it belongs in the roads table, 0 otherwise
+function roads(tags)
+    for k, v in pairs(tags) do
+        if roads_info[k] and roads_info[k][v] and roads_info[k][v].roads then
+            if not (k ~= 'railway' or tags.service) then
+                return 1
+            elseif not excluded_railway_service[tags.service] then
+                return 1
+            end
```

Comment:
For example, `highway=motorway + service=yard` will return 0. It's an unlikely combination but still likely not what you want to express.